infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/secret/FileSecretApplier.java

```diff
@@ -108,28 +122,38 @@ public void applySecret(KubernetesEnvironment env, RuntimeIdentity runtimeIdenti
               getOverridenComponentPath(component.get(), secret.getMetadata().getName());
         }
         final String componentMountPath = overridePathOptional.orElse(secretMountPath);
-        container
-            .getVolumeMounts()
-            .removeIf(vm -> Paths.get(vm.getMountPath()).equals(Paths.get(componentMountPath)));
-
-        secret
-            .getData()
-            .keySet()
-            .forEach(
-                secretFile ->
-                    container
-                        .getVolumeMounts()
-                        .add(
-                            new VolumeMountBuilder()
-                                .withName(volumeFromSecret.getName())
-                                .withMountPath(componentMountPath + "/" + secretFile)
-                                .withSubPath(secretFile)
-                                .withReadOnly(true)
-                                .build()));
+        // it's not possible to mount multiple volumes on same path on k8s older than 1.15, so we
+        // remove the existing mount here to replace it with new one.
+        if (k8sVersion.olderThan(1, 15)) {
+          container
+              .getVolumeMounts()
+              .removeIf(vm -> Paths.get(vm.getMountPath()).equals(Paths.get(componentMountPath)));
```

Comment:
What about just a mere overlap of paths? E.g. if the component mount path is a subpath or "superpath" of the existing mounts?

Reply:
Maybe it's more of a documentation issue than anything else...

Reply:
It should've been discussed here https://github.com/eclipse/che/pull/17167/files#diff-c28a5613d4bf66105777fb284783ba49R111
I will test the behavior and if needed, create another issue for it, ok ?

Reply:
I've tested again mounting secret `settings.xml` into `/home/user/.m2` and `/home/user/.m2/repository` from the devfile and it works ok on OpenShift 3.11